### DISCUSSION TITLE
计划任务备份应用_docker_全部失败问题

### DISCUSSION THREAD

**Discussion by JAINKRE** [1Panel-dev/1Panel#6214] [Q&A]
### 1Panel 版本

v1.10.13-lts

### 问题描述

计划任务备份全部docker应用，报错：

![PixPin_2024-08-22_22-05-26](https://github.com/user-attachments/assets/0c5c43f8-3d86-4f27-a24c-e9c1321c29a4)

如错误所示，找不到`it-tools`的备份路径，it-tools：

![20240822220658_rec_](https://github.com/user-attachments/assets/5794c0cc-11d1-4402-9d75-c2909b5069ab)

可以看到，应用商店安装的it-tools因为没有设置挂载路径，或者说这个容器就不需要挂载路径，但计划任务备份时还是去默认查找，就出错了。

建议优化这部分逻辑，可以考虑路径不存在直接跳过

### 重现步骤

如上

**Comment by JAINKRE** [comment]
还有一个问题，尝试卸载`it-tools`，会出现路径不存在错误，卸载不掉，只能强制删除

**Reply by wanghe-fit2cloud** [reply]
1、已安装列表显示 it-tools 应用运行正常，但 `./1panel/apps/it-tools` 目录不存在。是什么操作导致的？
> 该路径是 1Panel 安装应用的默认路径，如果应用运行正常，这个目录应该是存在的，不会被删除。

2、因为默认的应用安装目录已经不存在，所以需要强制删除。

**Reply by JAINKRE** [reply]
> 已安装列表显示 it-tools 应用运行正常，但 `./1panel/apps/it-tools` 目录不存在。是什么操作导致的？
> 
> > 该路径是 1Panel 安装应用的默认路径，如果应用运行正常，这个目录应该是存在的，不会被删除。

不清楚，it-tools可以正常访问，这个安装后就没动过

**Reply by wanghe-fit2cloud** [reply]
先排查一下吧，看看是什么原因导致的（1Panel 系统即使卸载应用，也会保留外层目录）。尝试定位一下具体是哪项操作删除了该目录。

**Reply by JAINKRE** [reply]
> 先排查一下吧，看看是什么原因导致的（1Panel 系统即使卸载应用，也会保留外层目录）。尝试定位一下具体是哪项操作删除了该目录。

ok，我先研究一下咋回事
